### PR TITLE
Add more vibe animations & clean up folders

### PR DESCRIPTION
**Pull request by matthewbeck**
Added animations
• Pulse
• Sparkle
• Fireworks
• Improved cursor spin
• Set foundation for calling random animation from array

### CODE REVIEW

**Comment by matthewbeck** [comment]
Big push with a lot of code. Might need a hand debugging vibe cursor spin. Every once in a while, it gets stuck as it loops from LED 180 to LED 0

File: animation/animation.ino [hunk 1]
```diff
@@ -0,0 +1,394 @@
...
+uint32_t last_buttoncheck = 0;
+uint32_t wait_buttoncheck = 10;
+
+uint32_t flashing_start_time = 0;
+
+// Function prototypes for the animations
+void rainbow_vibe();
+void vibe_cursor_spin();
+void vibe_pulse();
+void vibe_sparkle();
```
Comment (jazzz) [inline animation/animation.ino]: I'm not sure these are needed, though they don't hurt

File: animation/animation.ino [hunk 2]
```diff
@@ -0,0 +1,394 @@
...
+int chance_of_firework = 3; //set the likelihood of a firework going off in the firework animation
+bool go_firework = false; //variable to store whether firework should be shot off or not
+
+//Define the array of LEDs
+CRGB leds[NUM_LEDS];
+
+//Store animations in array
+void (*animationFunctions[])() = {rainbow_vibe, vibe_cursor_spin, vibe_pulse, vibe_sparkle};
+
+int numAnimations = sizeof(animationFunctions) / sizeof(animationFunctions[0]);
```
Comment (jazzz) [inline animation/animation.ino]: 😍

File: animation/animation.ino [hunk 3]
```diff
@@ -0,0 +1,394 @@
...
+
+
+//Setup steps go here
+void setup()
+{
+  
+  Serial.begin(115200);
+  delay(1000); // Put a small delay to ensure everything is setup properly
```
Comment (jazzz) [inline animation/animation.ino]: 100-300 milliseconds ought to be enough.

File: animation/animation.ino [hunk 4]
```diff
@@ -0,0 +1,394 @@
...
+  FastLED.addLeds<NEOPIXEL, DATA_PIN>(leds, NUM_LEDS);
+  pinMode(PIN_LED, OUTPUT);
+  digitalWrite(PIN_LED, HIGH); // Turn the built-in led on to show there is power
+  pinMode(BUTTON_PIN, INPUT_PULLUP);
+  fill_solid(leds, NUM_LEDS, 0x000000); //turn all leds off
+  FastLED.show();
+  
+  FastLED.setBrightness(100); //Set Brightness of LED strip
```
Comment (jazzz) [inline animation/animation.ino:83]: Great seeing this here to reduce wattage while testing. Once moved to final hardware make sure to bump the brightness up

File: animation/animation.ino [hunk 5]
```diff
@@ -0,0 +1,394 @@
...
+		leds[i] = CHSV(global_hue, 200, bri);
+	}
+  }
+    else {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(0, 0, 0);
+      } //If no button press, LEDs off
+    }
+    FastLED.show();
+
```
Comment (jazzz) [inline animation/animation.ino:136]:
```suggestion
void vibe_flash()
{
    for(int i = 0; i < NUM_LEDS; i++){
	uint8_t bri = (is_button_pressed) ? 255 : 0;
	leds[i] = CHSV(global_hue, 200, bri);
    }
    FastLED.show();
  
```

Calculate the value of `bri` based on whether `is_button_pressed`.

File: animation/animation.ino [hunk 6]
```diff
@@ -0,0 +1,394 @@
...
+///////////////////////////////////////
+// 3. Vibe Cursor Spin animation
+//// Animation displays a cursor of leds that cycle through the strip when button pressed. Sets all LEDs to black when button is depressed
+///////////////////////////////////////
+void vibe_cursor_spin()
+{
+  // a colored dot moving around a cicle, with fading trails
+  if(is_button_pressed){
+  fadeToBlackBy( leds, NUM_LEDS, 3);
```
Comment (jazzz) [inline animation/animation.ino:176]: This is clever. Never thought of doing it that way.

File: animation/animation.ino [hunk 7]
```diff
@@ -0,0 +1,394 @@
...
+  }
+  else {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(0, 0, 0);
+      } //If no button press, LEDs off
+    }
+  FastLED.show();
+  
+if (pos >= 255){
+  pos = 0;
```
Comment (jazzz) [inline animation/animation.ino]: There is a bug here:

`leds[pos] += CHSV( global_hue, 255, 255);`

The size of the `leds` array is NUM_LEDS, however `pos` could potentially be between [0,255]. When then the value is written to leds[255] this leads to memory corruption. 

1) make sure that data is never written to the `leds` array beyond index `NUM_LEDS-1`.

File: animation/animation.ino [hunk 8]
```diff
@@ -0,0 +1,394 @@
...
+  } //reset position of cursor back to 0 to spin it around the hoop
+
+  if(is_button_pressed == true & last_buttonpress == true){
+  pos++;
+  
+  }
```
Comment (jazzz) [inline animation/animation.ino]: Careful here: Incrementing the value after you've checked that the value is in bounds, means that the value could be bad when you go to use it next. 

Eg. lets say there is a value X which we only want to be between [0,3]

```
print(X) => 2
check(X) => 2
Inc(X) => 3
// Loop ----------
print(X) => 3
check(X) => 3
Inc(X) => 4
// loop ----------
print(X) => 4.  // Value is used out of bounds
check(X) => 0
Inc(X) => 1.  // Value 0 is never used becuase it is incremented immediately.
```

It's better to increment first, then check.
```
pos++;
if (pos >= NUM_LEDS){
  pos = 0;
}
```

File: animation/animation.ino [hunk 9]
```diff
@@ -0,0 +1,394 @@
...
+}
+
+///////////////////////////////////////
+// 4. Trunk charge animation
+//// Upon press, "charge up" the LEDs on the tree by turning on LEDs in series until reaching full bar
+///////////////////////////////////////
+void trunk_charge(){
+  // Charge up the trunk with a green light
+  if(is_button_pressed){
+  leds[pos] = CHSV(96, 237, 255);
```
Comment (jazzz) [inline animation/animation.ino:210]: Interesting Saturation value here(237) curious why that one? is this color matching a color palette someonewhere? If so leave a comment with where to find it so other contributors don't much with the color matching

File: animation/animation.ino [hunk 10]
```diff
@@ -0,0 +1,394 @@
...
+    }
+    //turn lights off after 2 seconds have passed since firework launch
+    else if(millis() - flashing_start_time >= 2000) {
+     for(int i = 0; i < NUM_LEDS; i++){
+    leds[i] = CHSV(0, 0, 0);
+    }
+    flashing = false;
+    }
+    }
+    else {
```
Comment (jazzz) [inline animation/animation.ino:301]: This is particularly hard to read. could use an `auto format`

File: animation/animation.ino [hunk 11]
```diff
@@ -0,0 +1,394 @@
...
+		    uint8_t bri = 255;
+		    leds[i] = CHSV(global_hue, 200, bri);
+      }
+	}
+  //Fade out the firework after being shown for one second
+    else if(millis() - flashing_start_time >= 1000 && millis() - flashing_start_time <= 2000) {
+      for(int i = 0; i < NUM_LEDS; i++){
+      leds[i] = CHSV(global_hue, 200, bri);
+      }
+      bri = bri - 5;
```
Comment (jazzz) [inline animation/animation.ino]: Careful these are using two different values of `bri`.

line 265 is using the variable defined on 264.
line 271 is using the variable defined on line 49.

File: animation/animation.ino [hunk 12]
```diff
@@ -0,0 +1,394 @@
...
+  }
+  last_buttonpress = is_button_pressed;
+}
+
+void addGlitter( fract8 chanceOfGlitter) 
+{
+  if( random8() < chanceOfGlitter) {
+    leds[ random16(NUM_LEDS) ] += CRGB::White;
+  }
+}
```
Comment (jazzz) [inline animation/animation.ino:353]: Might be helpful to add a reference to where this code came from so you can find it later

File: animation/animation.ino [hunk 13]
```diff
@@ -0,0 +1,394 @@
...
+    last_buttoncheck = millis();
+  }
+
+  // Update the GameState every N Milliseconds
+  if (last_draw + wait_draw < millis())
+  {
+    draw();
+    last_draw = millis();
+  }
+}
```
Comment (jazzz) [inline animation/animation.ino]: 👨‍🍳

**Review by jazzz — CHANGES_REQUESTED** [review]

**Pull request merged**